sphinx/util/fileutil.py

```diff
@@ -35,14 +51,13 @@ def copy_asset_file(source: str, destination: str,
         # Use source filename if destination points a directory
         destination = os.path.join(destination, os.path.basename(source))
 
-    if source.lower().endswith('_t') and context is not None:
+    if _template_basename(source) and context is not None:
         if renderer is None:
             from sphinx.util.template import SphinxRenderer
             renderer = SphinxRenderer()
 
         with open(source, encoding='utf-8') as fsrc:
-            if destination.lower().endswith('_t'):
-                destination = destination[:-2]
+            destination = _template_basename(destination) or destination
```

Comment:
(given earlier suggestion to make ``_template_basename`` always return *something*)
```suggestion
            destination = _template_basename(destination)
```